### PR TITLE
It's 4 am help me

### PR DESCRIPTION
**Pull request by Rick10101221**
Notes:
Queens Gambit is kinda cool def recommend

Work:
Integrated backend login functionality with frontend pages. 
Added and fixed other backend functions to perfectly fit needs of frontend onboarding.
Added comments and TODOs for backend team to view in api routes.

Accomplished:
Onboarding pages can now create users successfully, add to firebase authentication, create a new user object in firestore with user's uid as reference, and update username, firstname, and lastname with the exact same created user. pog

### CODE REVIEW

File: backend/routes/SignUp.js [hunk 1]
```diff
@@ -28,51 +44,58 @@ export function registerUser(email, password) {
...
+      first_name: '',
+      last_name: '',
+      profile_pic: '',
+      day_hug_count: 0,
+      current_streak: 0,
+      friends: [],
+      hugs: [],
+      chatrooms: [],
+      corkboard_id: newCorkboardRef.id,
+    }
```
Comment (evanjserrano) [inline backend/routes/SignUp.js:72]: We have Users/createUser so I'm not sure if we need this or not

Comment (evanjserrano) [inline backend/routes/SignUp.js:72]: We have Users/createUser so I don't know if this is needed

File: backend/routes/Users.js [hunk 2]
```diff
@@ -2,29 +2,22 @@
...
-  users.doc(username).set({
-    first_name: first,
-    last_name: last,
-  });
-} */
-
 const UsersAPI = {
 
-  user: Fire.auth().currentUser,
+  // user: firebase.auth().currentUser,
```
Comment (evanjserrano) [inline backend/routes/Users.js:15]: Why is this commented out? I think it will break functionality in some lines

**Review by evanjserrano — COMMENTED** [review]
See comments on Users.js and SignUp.js

**Comment by terryzfeng** [comment]
Approving SignUp.js and functions will be modified and copied over to Users.js

**Pull request merged**